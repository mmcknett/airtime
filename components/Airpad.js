import React, { useState } from 'react';
import { Button, ProgressViewIOS, Text, TouchableOpacity, View } from 'react-native';

export function Airpad(props) {
  const [nameTimes, setNameTimes] = useState([
    { name: 'Matt', time: 0 },
    { name: 'Amy', time: 0 }
  ]);

  const sum = nameTimes.reduce((a, b) => a.time + b.time);

  const addTime = (index, timeToAdd) => {
    const newNameTimes = nameTimes.slice();
    newNameTimes[index].time += timeToAdd;
    console.log('name times', newNameTimes);
    setNameTimes(newNameTimes);
  }

  resetAllTimes = () => {
    const newNameTimes = nameTimes.slice();
    newNameTimes.forEach(nameTime => { nameTime.time = 0; });
    console.log('name times', newNameTimes);
    setNameTimes(newNameTimes);
  }

  return (
    <>
      { 
        nameTimes.map(
          ({ name, time }, index) => 
            <NameTime
              key={ name }
              name={ name }
              fraction={ sum == 0 ? 0.01 : time / sum }
              addTime={ timeToAdd => addTime(index, timeToAdd) }
            />
        )
      }
      <Button title="Reset times" onPress={ resetAllTimes } />
    </>
  );
}

function NameTime(props) {
  const [lastPushTime, setLastPushTime] = useState(null);

  const onPressIn = () => {
    const now = new Date();
    setLastPushTime(now);
    console.log('Pressed:', now);
  }
  const onPressOut = () => {
    const timeToAdd = new Date() - lastPushTime;
    props.addTime(timeToAdd);
    setLastPushTime(null);
    console.log('Released:', timeToAdd);
  };

  return (
    <View
      style={{
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        marginBottom: 20
      }}
    >
      <ProgressViewIOS
        progress={ props.fraction }
        progressViewStyle='bar'
        style={{ width: 150, alignSelf: 'center' }}/>
      <TouchableOpacity
        title={ props.name }
        onPressIn={ onPressIn }
        onPressOut={ onPressOut }
        style={{ border: 1 }}
      >
        <Text style={{ fontSize: 24 }}>{ props.name }</Text>
      </TouchableOpacity>
    </View>
  );
}
