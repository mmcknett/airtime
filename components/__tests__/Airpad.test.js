import React from 'react';
import renderer from 'react-test-renderer';

import { Airpad } from '../Airpad';

it(`renders correctly`, () => {
  const tree = renderer.create(<Airpad />).toJSON();

  expect(tree).toMatchSnapshot();
});
