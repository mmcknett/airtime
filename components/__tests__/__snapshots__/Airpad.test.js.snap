// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 20,
      }
    }
  >
    <RCTProgressView
      progress={0.01}
      progressViewStyle="bar"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "alignSelf": "center",
            "width": 150,
          },
        ]
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "border": 1,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Matt
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 20,
      }
    }
  >
    <RCTProgressView
      progress={0.01}
      progressViewStyle="bar"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "alignSelf": "center",
            "width": 150,
          },
        ]
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "border": 1,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Amy
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Reset times
      </Text>
    </View>
  </View>,
]
`;
